test/Constraints/diagnostics.swift

```diff
@@ -188,6 +188,23 @@ func r17224804(_ monthNumber : Int) {
   let monthString = (monthNumber <= 9) ? ("0" + monthNumber) : String(monthNumber)
 }
 
+// <rdar://problem/33768959> QoI: invalid diagnostics when input to ternary operator does not exist
+func r33768959() {
+	enum E {
+        case LessThan
+        case EqualTo
+        // expected-note @+1 {{did you mean 'GreaterThan'?}}
+        case GreaterThan
+	}
+
+	func compare(_ lhs: Int, _ rhs: Int) -> E {
+        guard lhs != rhs else { return .EqualTo }
```

Comment:
Also, this part isn't really relevant to diagnostic emission and could be removed